### ISSUE TITLE
Support `==` on pd.Series

### ISSUE THREAD

**Issue by krivonogov** [tinyclues/fletcher#19]
```
df = pd.DataFrame({
    'fr': fr.FletcherArray([1, 2, 3]),
    'np': [1, 2, 3],
})

df['fr'] == 2
0    False
1    False
2    False
Name: int, dtype: bool

df['np'] == 2
0    False
1     True
2    False
Name: int, dtype: bool
```

**Comment by artemru** [comment]
it looks more like a bug as now...

**Comment by artemru** [comment]
```python
 df['fr'].astype(None) == 2                                                                                                          
Out[8]: 
0    False
1     True
2    False
Name: fr, dtype: bool
```
I think it's called with dtype='O' instead of None...

**Comment by artemru** [comment]
fixed by #27
